server.js

```diff
@@ -93,7 +93,10 @@ class Server {
   }
 }
 
-http.createServer(Server.requestListener).listen(port, () => {
-  // eslint-disable-next-line no-console
-  console.log(`Development server running: http://localhost:${port}`);
+const server = http.createServer(Server.requestListener);
+
+server.listen(process.env.PORT || 8080, () => {
+  const { address, port } = server.address();
+  Server.origin = `http://[${address}]:${port}`;
```

Comment:
We use introspection to generate the `origin` component necessary to qualify the path and leverage `URL` above for smarter path parsing.